CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+
+cmake_minimum_required(VERSION 3.15.1)
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules)
+
+project(SourceKit-LSP
+  LANGUAGES C Swift)
+
+set(CMAKE_Swift_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/swift)
+
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+if(CMAKE_VERSION VERSION_LESS 3.16 AND CMAKE_SYSTEM_NAME STREQUAL Windows)
+  set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+else()
+  set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+endif()
+
+find_package(dispatch QUIET)
+find_package(Foundation QUIET)
+find_package(TSC QUIET)
```

Comment:
Is this provided by SwiftPM or is it actually going to use the TSC repo?

Reply:
This is to use TSC from the tools-support-core repository.  All of the CI that I have going is trying to immediately use the tools-support-core version.

Reply:
This will block us testing cmake on Linux on ci.swift.org until TSC itself has PR testing that includes sourcekit-lsp.

Reply:
This is not a hard requirement - it will fall back to the system provided TSC if one is available.

Reply:
The cases I am concerned about are:

1) We make an API change to the copy of TSC in SwiftPM that requires a change in SourceKit-LSP.  If the cmake build uses the actual TSC repo but the swiftpm build uses TSC from SwiftPM we will need to make a lock-step change to TSC and SwiftPM.  This is a new requirement, but maybe it's "fine"?  That's a question for SwiftPM devs.

2) We make a change to TSC itself that breaks SourceKit-LSP.  Without PR testing for TSC that includes SourceKit-LSP, this will not be caught before merging.

Reply:
Okay, so you want to encourage the use of TSC from SwiftPM?  That is fine - if that is the preferred solution, my understanding as per @aciidb0mb3r was that we want to migrate to tools-support-core.

Reply:
I want us to migrate to TSC too, but I want to do that at the same time SwiftPM itself does.  I don't actually know what is holding that up.

Reply:
Well, the thing is, there is already a flag which controls that, and it is possible to build s-p-m against t-s-sc.

Reply:
> already a flag which controls that

Ah neat, so if we set `-DUSE_VENDORED_TSC=1` it will do that already?  If so, that's good enough for me.  If we add build using cmake that runs on ci.swift.org it should use that option to match how we build with swiftpm.

Reply:
Yeah, that would do it :)